eq-author-api/src/validation/customKeywords/validateRoutingLogicalAND.js

```diff
@@ -0,0 +1,140 @@
+const { groupBy } = require("lodash");
+const { getAnswerById } = require("../../../schema/resolvers/utils");
+const {
+  CURRENCY,
+  NUMBER,
+  PERCENTAGE,
+  UNIT,
+} = require("../../../constants/answerTypes");
+const createValidationError = require("../createValidationError");
+const { ERR_LOGICAL_AND } = require("../../../constants/validationErrorCodes");
+
+const getRHS = e =>
+  e.right && e.right.customValue && e.right.customValue.number;
+
+const areUnanswered = e => e.condition === "Unanswered";
+
+module.exports = ajv => {
+  ajv.addKeyword("validateRoutingLogicalAND", {
+    validate: function isValid(
+      otherFields,
+      { expressions },
+      fieldValue,
+      dataPath,
+      parentData,
+      fieldName,
+      questionnaire
+    ) {
+      const invalidAnswerIds = new Set();
+      const expressionsByAnswerId = groupBy(expressions, "left.answerId");
+      const potentialConflicts = Object.entries(expressionsByAnswerId).filter(
+        ([answerId, expressions]) =>
+          expressions.length > 1 &&
+          answerId &&
+          answerId.length &&
+          answerId.length > 0
```

Comment:
`answerId.length` && `answerId.length > 0` both do the same thing I think, if you're just checking to see if the `answerId` is populated?

Reply:
Cheers Tom, you're right. Will change.